features/mesh/dataplanes/DetailViewContent.feature

```diff
@@ -0,0 +1,101 @@
+Feature: Data Plane Proxies: Detail view content
+  Background:
+    Given the CSS selectors
+      | Alias            | Selector                                                  |
+      | detail-view      | [data-testid='data-plane-detail-view']                    |
+      | warnings         | [data-testid='data-plane-warnings']                       |
+      | overview-tab     | #overview-tab                                             |
+      | overview-content | #panel-0                                                  |
+      | insights-tab     | #insights-tab                                             |
+      | insights-content | #panel-1                                                  |
+      | status-cds       | $insights-content [data-testid='subscription-status-cds'] |
+      | status-eds       | $insights-content [data-testid='subscription-status-eds'] |
+      | status-lds       | $insights-content [data-testid='subscription-status-lds'] |
+      | status-rds       | $insights-content [data-testid='subscription-status-rds'] |
+    And the environment
+      """
+      KUMA_SUBSCRIPTION_COUNT: 2
+      """
+    And the URL "/config" responds with
+      """
+      body:
+        mode: global
+      """
+    # TODO: Use exact mocking here. It’s almost impossible to control the test data without exact mocking. For example, I need to set `disconnectTime: ''` just so that I don’t risk a faker-random disconnect time appearing in the last subscription.
```

Comment:
Why don't you only set the disconnectTime to ''? You shouldn't need to set all the other data.

Just so I understand, for the test, why is it necessary to have a disconnectTime of '' on the last subscription? Is it so a warning doesn't show up?

Reply:
Having `disconnectTime` set to `''` is a hack that avoids the default mock data randomly setting it. It ensures that I can control that this DPP always shows as offline.

I also need pretty much all the data here as it’s under test.

Reply:
Ah ok I see, so you are setting the subscription count to 2 to only get 2 subscriptions and then mocking out the final `disconnectTime` to be empty. Should it be null or undefined? Is that why you are saying it is a hack?

Reply:
I’m saying it’s a hack because what I would want is there to _not_ be a `disconnectTime` field to begin with because that’s the actual application scenario I want to test. We omit the field rather than it having a null value.

So, what I want is to leave it off and not have the mock merging mechanism populate it. That’s a really good example for when exact mocking is much, much easier to deal with.

Reply:
Right so you are saying it should be undefined? Not null or an empty ''?

Reply:
No, the field should be _absent_. Also, I shouldn’t need to know the default condition of the mock If I’m setting an array.

Reply:
> Also, I shouldn’t need to know the default condition of the mock If I’m setting an array.

No you shouldn't. That is why you should mock everything you test for instead of just relying on the default state like we used to. This is why the mocks try to give you random data. so you cannot depend on that.

Reply:
Yeah, but that’s precisely what’s not possible with the mock merging setup we have.

I cannot express "do not add a field to this object or array that’s not in my mock". I can only set it to something that has the same effect, but that’s hacky because it doesn’t reflect how our application actually works.

Reply:
We either need to add a way to set a field as undefined and therefore remove it, we should be able to add a YAML tag to do that.

 I think as it stands now you could set the count to 1 and then mock out 2 items, the last one having only the fields you want, which from what I remember from the code should also work.

It sounds like either way there are a few options here.